core/DataTable/Filter/SafeDecodeLabel.php

```diff
@@ -28,18 +36,37 @@ public function __construct($table)
         $this->columnToDecode = 'label';
     }
 
+    public static function shouldUrlDecodeValue(DataTable $table)
+    {
+        $archivedDate = $table->getMetadata(DataTable::ARCHIVED_DATE_METADATA_NAME);
+        if (empty($archivedDate)) {
+            return false;
+        }
+        $archivedDate = Date::factory($archivedDate);
+
+        $dateOfChange = Date::factory(self::DATE_OF_URLDECODE_CHANGE);
+
+        $shouldDecode = $archivedDate->isEarlier($dateOfChange);
+        return $shouldDecode;
+    }
+
     /**
      * Decodes the given value
      *
      * @param string $value
      * @return mixed|string
      */
-    public static function decodeLabelSafe($value)
+    public static function decodeLabelSafe($value, $shouldUrlDecode = true)
     {
         if (empty($value)) {
             return $value;
         }
-        $raw = urldecode($value);
+
+        $raw = $shouldUrlDecode ? urldecode($shouldUrlDecode) : $value;
```

Comment:
could this be a potential security issue?

Reply:
no longer relevant